// @owlbear-rodeo/sdk/lib/builders/EffectBuilder.js
import { GenericItemBuilder } from "./GenericItemBuilder";
export class EffectBuilder extends GenericItemBuilder {
    constructor(player) {
        super(player);
        this._item.name = "Effect";
        this._width = 0;
        this._height = 0;
        this._sksl = "";
        this._effectType = "VIEWPORT";
        this._uniforms = [];
        this._blendMode = "SRC_OVER";
    }
    width(width) {
        this._width = width;
        return this.self();
    }
    height(height) {
        this._height = height;
        return this.self();
    }
    sksl(sksl) {
        this._sksl = sksl;
        return this.self();
    }
    effectType(effectType) {
        this._effectType = effectType;
        return this.self();
    }
    uniforms(uniforms) {
        this._uniforms = uniforms;
        return this.self();
    }
    blendMode(blendMode) {
        this._blendMode = blendMode;
        return this.self();
    }
    build() {
        return Object.assign(Object.assign({}, this._item), { type: "EFFECT", width: this._width, height: this._height, sksl: this._sksl, effectType: this._effectType, uniforms: this._uniforms, blendMode: this._blendMode });
    }
}
