// @owlbear-rodeo/sdk/lib/builders/LightBuilder.js
import { GenericItemBuilder } from "./GenericItemBuilder";
export class LightBuilder extends GenericItemBuilder {
    constructor(player) {
        super(player);
        this._sourceRadius = 50;
        this._attenuationRadius = 12 * 150;
        this._falloff = 1.0;
        this._innerAngle = 360;
        this._outerAngle = 360;
        this._lightType = "PRIMARY";
        this._item.name = "Light";
        this._item.layer = "FOG";
        this._item.zIndex = 0;
        this._item.disableAutoZIndex = true;
    }
    sourceRadius(sourceRadius) {
        this._sourceRadius = sourceRadius;
        return this.self();
    }
    attenuationRadius(attenuationRadius) {
        this._attenuationRadius = attenuationRadius;
        return this.self();
    }
    falloff(falloff) {
        this._falloff = falloff;
        return this.self();
    }
    innerAngle(innerAngle) {
        this._innerAngle = innerAngle;
        return this.self();
    }
    outerAngle(outerAngle) {
        this._outerAngle = outerAngle;
        return this.self();
    }
    lightType(lightType) {
        this._lightType = lightType;
        return this.self();
    }
    build() {
        return Object.assign(Object.assign({}, this._item), { type: "LIGHT", sourceRadius: this._sourceRadius, attenuationRadius: this._attenuationRadius, falloff: this._falloff, innerAngle: this._innerAngle, outerAngle: this._outerAngle, lightType: this._lightType });
    }
}
